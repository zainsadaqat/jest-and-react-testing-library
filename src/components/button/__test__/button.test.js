import React from 'react';
import ReactDOM from 'react-dom';
import Button from './../button';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';

afterEach(cleanup);

it('Renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button></Button>, div);
});

// Testing Props
it('Renders button correctly', () => {
  const { getByTestId } = render(
    <Button label="Zain Sadaqat's Button"></Button>
  );
  expect(getByTestId('button')).toHaveTextContent("Zain Sadaqat's Button");
});

it('Renders button correctly second time', () => {
  const { getByTestId } = render(<Button label="My Button"></Button>);
  expect(getByTestId('button')).toHaveTextContent('My Button');
});

// Snapshot Testing
it('Matches Snapshot 1 ', () => {
  const tree = renderer.create(<Button label="click me"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
  // Press u to update the snapshot
});

it('Matches Snapshot 2 ', () => {
  const tree = renderer.create(<Button label="save me"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
  // Press u to update the snapshot
});
