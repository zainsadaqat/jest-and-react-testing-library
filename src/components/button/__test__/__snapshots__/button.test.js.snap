// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1  1`] = `
<div
  className="button-style"
  data-testid="button"
>
  click me
</div>
`;

exports[`Matches Snapshot 2  1`] = `
<div
  className="button-style"
  data-testid="button"
>
  save me
</div>
`;
